src/maplayer.c

```diff
@@ -105,7 +105,15 @@ int msLayerRestoreFromScaletokens(layerObj *layer) {
     layer->tileitem = layer->orig_st->tileitem;
   }
   if (layer->orig_st->filter) {
-    msLoadExpressionString(&(layer->filter), layer->orig_st->filter);
+    if (layer->filter.type == MS_EXPRESSION) {
+      char *tmpval = (char *)msSmallMalloc(sizeof(char) *
```

Comment:
Although it is fine here, for the sake of the relief of our souls, I would store the size in a variable and then use it with snprintf()

something like
```
const size_t tmpval_size = strlen(layer->filter.string) + 3;    // you can drop sizeof(char). It is always 1
char *tmpval = (char *)msSmallMalloc(tmpval_size);
snprintf(tmpval, tmpval_size, "(%s)", layer->filter.string);
```
same below